tests/fixtures/middleware-i18n/middleware.js

```diff
@@ -17,11 +20,9 @@ export async function middleware(request) {
   }
 
   if (url.pathname.startsWith('/link/rewrite-me')) {
-    const rewriteUrl = new URL(
-      url.pathname.replace('/link/rewrite-me', '/link/rewrite-target'),
-      url,
-    )
-    return NextResponse.rewrite(rewriteUrl, {
+    // use NextUrl to rewrite the URL as it does preserve the locale
+    url.pathname = url.pathname.replace('/link/rewrite-me', '/link/rewrite-target')
+    return NextResponse.rewrite(url, {
```

Comment:
Note: previous rewrite was just losing locale url due to using just `NextURL.pathname` to construct new `URL` effectively losing locale context in the process. 

This change keeps using `NextURL` and just mutate `pathname` portion of it which preserve `locale` context and resulting in seemingly working as expected rewrites